src/index.js

```diff
@@ -106,8 +121,18 @@ async function main() {
 
   // TODO: how do we know the task happens? Could we stream reading events on Mangata side?
   console.log(`\n4. waiting for XCM events on Mangata side ...`);
-
   await listenEvents(mangataHelper.api);
+
+  console.log(`\nWaiting 20 seconds before reading new chain states ...`);
+  await delay(20000);
```

Comment:
This took me for a while to figure out. Without waiting 20 seconds the reserved balance of LP doesn’t change. I don’t understand why there’s such a delay in getBalance() call. It must have something to do with Mangata’s Js API or code implementation, that there’s a delay between event happened and chain state query result update? 

Maybe the new chain state is not finalized and retrievable until the next block?

Reply:
This is possible. There are often api delays related to finalization that are not always transparent.